Apps/Sandcastle/gallery/Particle System Tails.html

```diff
@@ -0,0 +1,260 @@
+        red : 0.5,
+        green : 0.5,
+        blue : 0.7,
+        alpha : 0.5
+    }, {
+        red : 0.5,
+        green : 0.5,
+        blue : 0.5,
+        alpha : 0.9
+    }],
+
+    particleSystems : []
+};
+
+// Creating the Particles
+function create(usingOptions, newLocation, color, iteration) {
+    var cartesianOrigPosition = Cesium.Cartographic.toCartesian(newLocation);
+    var emitterModelMatrix = Cesium.Matrix4.fromTranslation(cartesianOrigPosition, new Cesium.Matrix4());
+
+    var iterOffset = iteration;
+    var force = function(particle) {
+        var cartesianPosition = particle.position;
+        var downDir = Cesium.Cartesian3.normalize(cartesianPosition, new Cesium.Cartesian3());
+
+        cartesianPosition.x -= downDir.x;
+        cartesianPosition.y -= downDir.y;
+        cartesianPosition.z -= downDir.z;
+
+        var cartographicPosition = Cesium.Cartographic.fromCartesian(cartesianPosition);
+
+        var angle = Cesium.Math.PI * 2.0 * iterOffset / usingOptions.numberOfParticles;
+        iterOffset += usingOptions.iterOffset;
+        cartographicPosition.longitude += Math.cos(angle) * usingOptions.cartographicStep;
+        cartographicPosition.latitude += Math.sin(angle) * usingOptions.cartographicStep;
+
+        cartesianPosition = Cesium.Cartographic.toCartesian(cartographicPosition);
+
+        particle.position = cartesianPosition;
+    };
+
+    var item = viewer.scene.primitives.add(new Cesium.ParticleSystem({
+        image : getImage(),
+        startColor : color,
+        endColor : color.withAlpha(0.0),
+        life : 3.5,
+        speed : 0.00005,
+        width : 15.0,
+        height : 15.0,
+        rate : 30.0,
+        emitter : new Cesium.CircleEmitter(0.1),
+        bursts : [ ],
+        lifeTime : 0.1,
+        forces : [force],
+        modelMatrix : modelMatrix,
+        emitterModelMatrix : emitterModelMatrix
+    }));
+    usingOptions.particleSystems.push(item);
+}
+
+function runParticles(usingOptions) {
+    var scratchAngleForOffset = 0.0;
+    var scratchOffset = Cesium.Cartesian3.clone(centerLocation);
+    for (var i = 0; i < usingOptions.numberOfParticles; ++i) {
+        scratchAngleForOffset = Math.PI * 2.0 * i / usingOptions.numberOfParticles;
+        scratchOffset.x += usingOptions.baseRadius * Math.cos(scratchAngleForOffset);
+        scratchOffset.y += usingOptions.baseRadius * Math.sin(scratchAngleForOffset);
+
+        var cartographicOffset = Cesium.Cartographic.fromCartesian(scratchOffset);
+
+        var color = Cesium.Color.fromRandom(usingOptions.colorOptions[i % usingOptions.colorOptions.length]);
+        create(usingOptions, cartographicOffset, color, i);
+    }
+}
+
+// initializing particleSystems arrays
+runParticles(rocketOptions);
+runParticles(cometOptions);
+
+function showAll(particleSystemsArray, showing) {
+    var i = 0;
+    while (i < particleSystemsArray.length) {
+        particleSystemsArray[i].show = showing;
+        ++i;
+    }
```

Comment:
Any particular reason we're not using a for loop here?

Also Cesium coding style is to save this value before the loop:
```
var length = particleSystemsArray.length;
for (var i = 0; i < length; ++i) {
```